src/actions/wallet.js

```diff
@@ -97,6 +97,16 @@ class ActionsWallet {
         this.t2342 = setTimeout(() => this.getNewAddress(), RETRY_DELAY);
       });
   }
+
+  async getIP() {
+    try {
+      const request = await fetch('https://api.ipify.org?format=json');
+      const response = await request.json();
+      this._store.IP = response.ip;
+    } catch (e) {
+      throw new Error('Error fetching IP');
+    }
```

Comment:
Seems like the app would crash if IP detection via the api fails. Either we log `e` to the console or handle it some other way.